herd/ASLSem.ml

```diff
@@ -160,8 +161,35 @@ module Make (C : Config) = struct
       | 128 -> MachSize.S128
       | _ -> Warn.fatal "Cannot access a register with size %s" (V.pp_v v)
 
-    let to_bv = M.op1 (Op.ArchOp1 ASLValue.ASLArchOp.ToBV)
-    let to_int = M.op1 (Op.ArchOp1 ASLValue.ASLArchOp.ToInt)
+    let to_bv v =
+      match v with
+      | V.Val (Constant.Concrete (ASLScalar.S_BitVector _)) -> return v
+      | V.Var _ | V.Val (Constant.Concrete _) ->
+          M.op1 (Op.ArchOp1 ASLValue.ToBV) v
+      | v ->
+          let v' = V.fresh_var () in
+          M.restrict M.VC.[ Assign (v', Unop (Op.ArchOp1 ASLValue.ToBV, v)) ]
+          >>! v'
```

Comment:
If seen that M.op1 takes care of most of those cases. Wouldn't the following code be functionaly equivalent:
```suggestion
    let to_bv v = M.op1 (Op.ArchOp1 ASLValue.ToBV) v
```
And if there are differences are they justified?

Reply:
Done in cac5b6aa5659f8ccdb3597f39b96c99099c4ea1e